packages/@aws-cdk/core/lib/nested-stack.ts

```diff
@@ -109,8 +105,7 @@ export class NestedStack extends Stack {
 
     this._parentStack = parentStack;
 
-    // @deprecate: remove this in v2.0 (redundent)
```

Comment:
Why the comment is removed?

Reply:
@foriequal0 We decided to keep this hierarchy to avoid breaking users (this would cause all logical IDs to be changed).

What is the reason you are asking?

Reply:
Okay I understand that. I was a little bit annoyed that it makes already long resource name very long and repetitive.
For e.g., `@aws-cdk/aws-eks.ClusterResourceProvider`'s resource name is `awscdkawseksClusterResourceProviderNestedStackawscdkawseksClusterResourceProviderNestedStackResource9827C454`
It is derived from its path
```js
[
  // from id of parentScope
  "@aws-cdk/aws-eks.ClusterResourceProvider.NestedStack",
  // from id of this.resource
  "@aws-cdk/aws-eks.ClusterResourceProvider.NestedStackResource"
]
```
I hoped that it will be `["@aws-cdk/aws-eks.ClusterResourceProvider", "NestedStackResource"]` in v2.0.

Reply:
Understood, but sadly this will cause too much disruption. Locally, you can opt to use [`stack.renameLogicalId()`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Stack.html#renamewbrlogicalwbridoldid-newid) to explicitly set the logical ID of resources in your stack.